internal/api/sms_provider/textlocal.go

```diff
@@ -86,17 +86,17 @@ func (t *TextlocalProvider) SendSms(phone string, message string) (string, error
 		return "", derr
 	}
 
-	if len(resp.Errors) > 0 {
-		return "", errors.New("textlocal error: Internal Error")
-	}
-
 	messageID := ""
 
 	if resp.Status != "success" {
 		if len(resp.Messages) > 0 {
 			messageID = resp.Messages[0].MessageID
 		}
 
+		if resp.Errors[0].Code == textLocalTemplateErrorCode {
```

Comment:
/nit I know it's guaranteed to be a `success` but out of an abundance of caution can we consider adding a check to see that there are entries in the array?
```suggestion
		if len(resp.Errors) > 0 && resp.Errors[0].Code == textLocalTemplateErrorCode {
```

Reply:
Otherwise, seems fine to me. Thanks for the PR!

Reply:
done, thanks @J0